src/client/dfuse/ops/write.c

```diff
@@ -24,18 +24,49 @@
 #include "dfuse_common.h"
 #include "dfuse.h"
 
+static void
+dfuse_cb_write_complete(struct dfuse_event *ev)
+{
+	if (ev->de_ev.ev_error == 0)
+		DFUSE_REPLY_WRITE(ev, ev->de_req, ev->de_len);
+	else
+		DFUSE_REPLY_ERR_RAW(ev, ev->de_req,
+				    daos_der2errno(ev->de_ev.ev_error));
+	D_FREE(ev->de_buff);
+}
+
 void
 dfuse_cb_write(fuse_req_t req, fuse_ino_t ino, const char *buff, size_t len,
 	       off_t position, struct fuse_file_info *fi)
 {
 	struct dfuse_obj_hdl		*oh = (struct dfuse_obj_hdl *)fi->fh;
-	d_iov_t				iov = {};
-	d_sg_list_t			sgl = {};
+	struct dfuse_projection_info	*fs_handle = fuse_req_userdata(req);
 	int				rc;
+	struct dfuse_event		*ev;
+
+	D_ALLOC_PTR(ev);
+	if (ev == NULL)
+		D_GOTO(err, rc = ENOMEM);
+
+	D_ALLOC(ev->de_buff, len);
+	if (ev->de_buff == NULL)
+		D_GOTO(err, rc = ENOMEM);
+
+	memcpy(ev->de_buff, buff, len);
```

Comment:
im thinking that we should really see the effect of this before landing the patch.
Do you think there is any value to this patch if performance actually takes a hit?